src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CspParameters.cs

```diff
@@ -49,11 +49,7 @@ public CspProviderFlags Flags
         }
 
         [CLSCompliantAttribute(false)]
-        public SecureString KeyPassword
-        {
-            get { return null; }
-            set { throw new PlatformNotSupportedException(); }
-        }
+        public SecureString KeyPassword { get; set; }
```

Comment:
If set isn't going to throw we need to make it work.  I can provide more data in the morning; but this should definitely not be merged in this state.